src/ParallelAlgorithms/Actions/MutateApply.hpp

```diff
@@ -0,0 +1,53 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <tuple>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace tuples {
+template <typename...>
+class TaggedTuple;
+}  // namespace tuples
+
+namespace Parallel {
+template <typename Metavariables>
+class ConstGlobalCache;
+}  // namespace Parallel
+/// \endcond
+
```

Comment:
We should probably have a namespace for the parallel algs. maybe `pa` or `palg` since they're short?

Reply:
I was considering that, but then didn't see the point of wrapping all of those components that we'll place in `ParallelAlgorithms` in a namespace. They don't really have anything in common, except for the fact that they are built on top of the parallel backend, but that's basically our entire code. So I'd suggest to just go with `Interpolation::` etc. top-level namespaces.

For this reason I also placed this action in a top-level `Actions` namespace. All actions are built the parallel backend, so prepending a namespace is kind of tautologic. Thoughts?

Reply:
I agree for `Actions`. For ones like interpolation I was thinking it would be nice to keep the serial and parallel interpolation code in a separate namespace. What do you think?

Reply:
I don't know enough about the interpolation code, so will instead refer to the DG code. I kind of see the whole DG code that's currently in `NumericalAlgorithms/DG` as a component of our code that is built with our parallel architecture in mind, so wouldn't split parts of it into a serial namespace. I would only keep code in `NumericalAlgorithms` that really has nothing to do with parallelism, like a serial root finder, the spectral stuff, the linear operators, ... basically it's `Utilities` but for more advanced algorithms that may require significant amount of code.

Reply:
I think this is a case-by-case thing. I agree that for some things it makes sense to have them be in the same namespace, but I think for others it might not. Let's leave that for when individual parts are moved over :)

Reply:
yes let's do that